### ISSUE TITLE
About Model Inference Latency in production environments

### ISSUE THREAD

**Issue by mushan09** [facebookresearch/FixRes#26]
Hi,

I am very interested in fixres and consider the possibility of using it in actual production environments. I testd different image size without fixres or another finetune, the only difference is that I resize the input test image to a different size.

My test environment is tensorrt7.0.0.11, cuda 10.1, NVIDIA T4. Below are some of my test conclusions

test_image_size | batch_size | accuracy | lantency
-- | -- | -- | --
224 | 1 | 75.49 | 0.5452
336 | 1 | 76.92 | 0.7595
384 | 1 | 76.87 | 0.7859

It seems that the cost of fixres is greater than the benefit of accuracy. 
So I want to ask what are the best practices for using fixres in actual business scenarios

Really appreaciate that you could reply!

**Comment by TouvronHugo** [comment]
Hi @mushan09 ,

If I understand correctly you trained your model with a certain resolution and you evaluated it at different resolutions without using FixRes, but the problem is that it increases the latency time.
If latency is an important point and you want for example to keep an evaluation at resolution 224x224.
I suggest you train your model with a lower resolution (for instance 160x160) and then use FixRes to adapt it to the 224x224 test resolution.
I hope I have answered your question, don't hesitate if you have other questions,

Best,

Hugo

**Comment by mushan09** [comment]
Hi, Hugo

Thank you for your reply.
In your paper, Figure 5 shows that train model with 224x224 resolution and then use FixRes to adapt it to the 384x384 test resolution is better than using 224x224 for both training and testing.
But training with 128x128 and adapting to 224x224 is almost equal to use 224x224 for both training and testing.

I want to improve the accuracy without increasing the inference latency, but training with lower resolution does not seem to be very helpful to the accuracy, but it can accelerate the training procedure. So I am not sure if the fixres meets my expectations.

I will do some experimentsto verify, if you have some experimental data to share, I would be very grateful.

Sincerely,

Steve

**Comment by TouvronHugo** [comment]
Hi @mushan09 ,

With ResNet-50 train on ImageNet at resolution 160x160 + FixRes at resolution 224x224 we get 77.6% (Appendix Table 7).
This is an improvement of 0.6 compared to the baseline (train 224x224 -> test 224x224). 

I hope this helps,
Best,

Hugo
